packages/context/src/context.ts

```diff
@@ -161,8 +167,23 @@ export class Context extends EventEmitter {
   private setupEventHandlersIfNeeded() {
     if (this.notificationQueue != null) return;
 
-    this.addParentEventListener('bind');
-    this.addParentEventListener('unbind');
+    if (this._parent != null) {
+      /**
+       * Add an event listener to its parent context so that this context will
+       * be notified of parent events, such as `bind` or `unbind`.
+       */
+      this.parentEventListener = (
+        binding: Readonly<Binding<unknown>>,
+        context: Context,
+        event: string,
+      ) => {
+        this.handleParentEvent(binding, context, event);
+      };
+
+      // Listen on the parent context events
+      this._parent.on('bind', this.parentEventListener!);
+      this._parent.on('unbind', this.parentEventListener!);
```

Comment:
Should we also listen on the `changed` event introduced by https://github.com/strongloop/loopback-next/pull/4430 and propagate it via `parentEventListener`?

Reply:
Please note `changed` events are only emitted by bindings at the moment. We're yet to introduce `bindingChanged` event at context level.